core/tests/protractor_utils/editor.js

```diff
@@ -377,9 +386,10 @@ var addParameterChange = function(paramName, paramValue) {
 // should be specified after the ruleName as additional arguments. For example
 // with interaction 'NumericInput' and rule 'Equals' then there is a single
 // parameter which the given answer is required to equal.
-var _selectRule = function(ruleElement, interactionId, ruleName) {
+var _selectRule = function(ruleElement, interactionId, ruleName,
+  skipParameters) {
```

Comment:
It looks like to me like you need to split out the functionality of _selectRule() into the part that chooses the rule and the part that enters its parameters. To do this I would modify the existing _selectRule() to just do the choosing part, and then have a new _setRuleParameters() function to put in the parameters. You can then modify all the pre-existing places that make use of _selectRule() to now call both in sequence. That should be unproblematic as _selectRule() is local to this file.